src/main/groovy/com/cloudcard/photoDownloader/TouchNetClient.groovy

```diff
@@ -111,26 +182,38 @@ class TouchNetClient {
         return response.success
     }
 
-    boolean accountPhotoApprove(String sessionId, String accountId, String photoBase64) {
-        Map request = [
-            SessionID: sessionId,
-            AccountID: accountId,
-            PhotoBase64: photoBase64,
-            ForcePrintedFlag: false
-        ]
+    private static boolean isRateLimitError(String message) {
+        if (!message) return false
 
-        TouchNetResponse response = doApiRequest("Account Photo Approve", "account/photo/approve", request)
+        def lower = message.toLowerCase()
 
-        return response.success
+        return lower.contains("api calls") || lower.contains("quota exceeded") || lower.contains("maximum admitted")
     }
 }
 
 class TouchNetResponse {
     boolean success
     Object json
```

Comment:
Yeah, I'd like to have more structured types here and start using more objectmapper...